packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -2,33 +2,80 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React, { useState } from 'react';
-import { RouteComponentProps } from '@reach/router';
+import React, { useEffect, useState } from 'react';
+import { navigate, RouteComponentProps, useLocation } from '@reach/router';
+import { REACT_ENTRYPOINT } from '../../../constants';
+import {
+  AuthUiErrors,
+  composeAuthUiErrorTranslationId,
+} from '../../../lib/auth-errors/auth-errors';
+import { logViewEvent, usePageViewEvent } from '../../../lib/metrics';
 import { FtlMsg } from 'fxa-react/lib/utils';
-import { useFtlMsgResolver } from '../../../models/hooks';
-import { usePageViewEvent } from '../../../lib/metrics';
-// import { useAlertBar } from '../../models';
+import {
+  useAccount,
+  useAlertBar,
+  useFtlMsgResolver,
+} from '../../../models/hooks';
+import AppLayout from '../../../components/AppLayout';
+import Banner, { BannerType } from '../../../components/Banner';
+import CardHeader from '../../../components/CardHeader';
 import FormVerifyCode, {
   FormAttributes,
 } from '../../../components/FormVerifyCode';
-import { REACT_ENTRYPOINT } from '../../../constants';
 import { MailImage } from '../../../components/images';
-import CardHeader from '../../../components/CardHeader';
-
-// email will eventually be obtained from account context
-export type ConfirmSignupCodeProps = { email: string };
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import { Newsletter } from '../../../components/ChooseNewsletters/newsletters';
 
 export const viewName = 'confirm-signup-code';
 
-const ConfirmSignupCode = ({
-  email,
-}: ConfirmSignupCodeProps & RouteComponentProps) => {
+export type RouterState = {
+  email: string;
+  newsletters?: Newsletter[];
+};
+
+interface BannerOptions {
+  visible: boolean;
+  type?: BannerType;
+  message?: string;
+}
+
+const ConfirmSignupCode = (_: RouteComponentProps) => {
   usePageViewEvent(viewName, REACT_ENTRYPOINT);
 
-  // const alertBar = useAlertBar();
   const ftlMsgResolver = useFtlMsgResolver();
-  const [code, setCode] = useState<string>('');
+  const alertBar = useAlertBar();
+  const account = useAccount();
   const [codeErrorMessage, setCodeErrorMessage] = useState<string>('');
+  const [resendCodeCount, setResendCodeCount] = useState<number>(0);
+  const [clearMessages, setClearMessages] = useState<boolean>(false);
+
+  let { state = {} } = useLocation();
+  if (!state) {
+    state = {};
+  }
```

Comment:
You shouldn't need to do it this way, [see how I did this here](https://github.com/mozilla/fxa/pull/14933/files#diff-99ad6a399825850b7a5cdb423368271e810e574014e2a70c35864c2653d0496fR70-R72).